eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -173,6 +179,23 @@ jobs:
           node eng/tools/rush-runner.js lint "${{parameters.ServiceDirectory}}"
         displayName: "Lint libraries"
 
+      - script: |
+          node eng/tools/rush-runner.js audit "${{parameters.ServiceDirectory}}"
+        condition: and(succeeded(), eq(variables['RunNpmAudit'], 'true'))
+        displayName: "Audit libraries"
+
+      - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+        # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
+        # builds should be sufficient.
+        condition: and(succeededOrFailed(), ne(variables['Build.Reason'], 'PullRequest'))
+        displayName: "Component Detection"
+
+      # Unlink node_modules folders to significantly improve performance of subsequent tasks
+      # which need to walk the directory tree (and are hardcoded to follow symlinks).
+      - script: |
```

Comment:
I assume the big issue is that the node_modules folders is next to every project. Have we considered trying to redirect the node_modules folder to have one shared one for all projects? Seems like there might be some ways to configure that in an npmrc config https://docs.npmjs.com/files/folders#tl-dr.

Reply:
@weshaggard: This is all managed by Rush.  Rush creates a `node_modules` folder in the repo root and installs all dependencies used by all packages into this folder.  Then, it creates a `node_modules` folder in each package folder, but in these folders everything is symlinked back to the root `node_modules` folder.  This allows every package to see only the dependencies they expect, while at the same time avoiding duplication of dependencies on disk.  I don't think we want to try changing how this works, as it's pretty fundamental to Rush.